### PR TITLE
Admin can see all sources (and cache user in redis)

### PR DESCRIPTION
**Pull request by simonvadee**
user is cached in redis on login/signup
user is put in context for every request (if auth token is ok)

update the `sources` resolver so that admins can see all sources

### CODE REVIEW

File: server/src/resolvers/Source.ts [hunk 1]
```diff
@@ -57,12 +57,13 @@ export const sources: FieldResolver<'Query', 'sources'> = async (
...
-  const userId = getUserId(ctx)
-  const accesses = await ctx.prisma.accessControl.findMany({
-    where: { user: { id: userId } },
-    include: { source: true },
+  const { role, id } = ctx.user!
+
+  if (role === 'ADMIN') return ctx.prisma.source.findMany()
+
+  return ctx.prisma.source.findMany({
+    where: { accessControls: { some: { user: { id } } } },
```
Comment (Jasopaum) [inline server/src/resolvers/Source.ts]: Cool, didn't know we could do that 👍

File: server/src/cache.ts [hunk 2]
```diff
@@ -5,6 +5,7 @@ import { REDIS_URL } from './constants'
 
 let rcache: RedisClient
 let get: (arg: string) => Promise<string>
+let del: (arg: string) => Promise<number>
```
Comment (Jasopaum) [inline server/src/cache.ts]: You finally didn't use this, did you?

Comment (simonvadee) [reply server/src/cache.ts]: no I didn't but now it's there when we need it :)

Comment (Jasopaum) [reply server/src/cache.ts]: What about YAGNI? ;)

Comment (simonvadee) [reply server/src/cache.ts]: what do you think the french government thinks about YAGNI now that we're out of masks and respirators ?

Comment (simonvadee) [reply server/src/cache.ts]: covid-19 should be an inspiration to all developers around the world

Comment (Jasopaum) [reply server/src/cache.ts]: ❤️

**Pull request merged**